ts/Extensions/DashboardPlugins/DataGridComponent.ts

```diff
@@ -342,38 +377,89 @@ class DataGridComponent extends Component<DataGridComponent.ChartComponentEvents
  * */
 
 namespace DataGridComponent {
+
+    /* *
+     *
+     *  Declarations
+     *
+     * */
+
+    /** @internal */
     export type ComponentType = DataGridComponent;
 
+    /** @internal */
     export type ChartComponentEvents = JSONEvent | Component.EventTypes;
 
+    /** @internal */
     export type JSONEvent = Component.Event<
     'toJSON' | 'fromJSON',
     {
         json: ClassJSON;
     }
     >;
 
+    /**
+     * Options to control the DataGrid component.
+     */
     export interface DataGridOptions
         extends Component.ComponentOptions,
         EditableOptions {
+
+        /**
+         * The style class to add to the rendered data grid container.
+         */
         dataGridClassName?: string;
+
+        /**
+         * The identifier for the rendered data grid container.
+         */
         dataGridID?: string;
+
+        /**
+         * Callback to use when a change in the data grid occures.
+         */
         onUpdate: typeof DataGridComponent.onUpdate
     }
 
+    /**
+     * Options to control the DataGrid component. These options can be changed
+     * in edit mode of a dashboard.
+     */
     export interface EditableOptions extends Component.EditableOptions {
-        dataGridOptions?: DataGridOptions;
+
+        /**
+         * Generic options to adjust behavor and styling of the rendered data
```

Comment:
I think @sebastianbochan already started working on describing `DataGridComponent`.
Please check with him in order to avoid conflicts.